DICOM/src/Microsoft.Health.Dicom.Anonymizer.Core/Processors/CryptoHashProcessor.cs

```diff
@@ -0,0 +1,94 @@
+    /// </summary>
+    public class CryptoHashProcessor : IAnonymizerProcessor
+    {
+        private readonly CryptoHashFunction _cryptoHashFunction;
+        private static readonly HashSet<string> _supportedVR = Enum.GetNames(typeof(CryptoHashSupportedVR)).ToHashSet(StringComparer.InvariantCultureIgnoreCase);
+
+        public CryptoHashProcessor(JObject settingObject)
+        {
+            EnsureArg.IsNotNull(settingObject, nameof(settingObject));
+
+            var settingFactory = new AnonymizerSettingsFactory();
+            var cryptoHashSetting = settingFactory.CreateAnonymizerSetting<CryptoHashSetting>(settingObject);
+            _cryptoHashFunction = new CryptoHashFunction(cryptoHashSetting);
+        }
+
+        public void Process(DicomDataset dicomDataset, DicomItem item, ProcessContext context = null)
+        {
+            EnsureArg.IsNotNull(dicomDataset, nameof(dicomDataset));
+            EnsureArg.IsNotNull(item, nameof(item));
+
+            if (item is DicomStringElement)
+            {
+                var encryptedValues = ((DicomStringElement)item).Get<string[]>().Select(GetCryptoHashString);
+                dicomDataset.AddOrUpdate(item.ValueRepresentation, item.Tag, encryptedValues.ToArray());
+            }
+            else if (item is DicomOtherByte)
+            {
+                var valueBytes = ((DicomOtherByte)item).Get<byte[]>();
+                var encryptesBytes = _cryptoHashFunction.ComputeHmacSHA256Hash(valueBytes);
+                dicomDataset.AddOrUpdate(item.ValueRepresentation, item.Tag, encryptesBytes);
+            }
+            else if (item is DicomFragmentSequence)
+            {
+                var enumerator = ((DicomFragmentSequence)item).GetEnumerator();
+
+                var element = item.ValueRepresentation == DicomVR.OW
+                    ? (DicomFragmentSequence)new DicomOtherWordFragment(item.Tag)
+                    : new DicomOtherByteFragment(item.Tag);
+
+                while (enumerator.MoveNext())
+                {
+                    element.Fragments.Add(new MemoryByteBuffer(_cryptoHashFunction.ComputeHmacSHA256Hash(enumerator.Current.Data)));
+                }
+
+                dicomDataset.AddOrUpdate(element);
+            }
+            else
+            {
+                throw new AnonymizationOperationException(DicomAnonymizationErrorCode.UnsupportedAnonymizationMethod, $"CryptoHash is not supported for {item.ValueRepresentation}.");
+            }
+        }
+
+        public bool IsSupportedVR(DicomItem item)
+        {
+            EnsureArg.IsNotNull(item, nameof(item));
+
+            return _supportedVR.Contains(item.ValueRepresentation.Code) || item is DicomFragmentSequence;
+        }
+
+        public string GetCryptoHashString(string input)
+        {
+            EnsureArg.IsNotNull(input, nameof(input));
+
+            var resultBytes = _cryptoHashFunction.ComputeHmacSHA256Hash(Encoding.UTF8.GetBytes(input));
```

Comment:
Can we add an interface in cryptoHashFunction: ComputeHash(string base64Text, CryptoHashMethodType type)?
As you already defined the function types of SHA128, SHA256